src/org/jetbrains/java/decompiler/modules/decompiler/IfHelper.java

```diff
@@ -313,6 +312,7 @@ else if (elsebranch.succs.size() == 1) {
 
           StatEdge newifedge = new StatEdge(StatEdge.TYPE_REGULAR, firstif.getFirst(), second);
           firstif.getFirst().addSuccessor(newifedge);
+          firstif.setIfEdge(newifedge);
```

Comment:
This has also been added on my branch, when we merge, we should probably check that we don't have this being set twice. It wouldn't do any harm but it would look a bit weird :p